editor/src/components/inspector/sections/component-section/data-reference-cartouche.tsx

```diff
@@ -99,14 +103,29 @@ export const DataReferenceCartoucheControl = React.memo(
       [dispatch, props.renderedAt],
     )
 
-    const propertyPath =
+    const maybePropertyPath =
       props.renderedAt.type === 'element-property-path'
         ? props.renderedAt.elementPropertyPath.propertyPath
-        : props.renderedAt.type === 'child-node'
-        ? PP.create('children')
-        : assertNever(props.renderedAt)
+        : null
 
-    const variableNamesInScope = useVariablesInScopeForSelectedElement(elementPath, propertyPath)
+    const controlDescriptions = usePropertyControlDescriptions(maybePropertyPath)
+    const currentPropertyValue = usePropertyValue(elementPath, maybePropertyPath)
+
+    const matcher = React.useMemo(
+      () =>
+        props.renderedAt.type === 'element-property-path'
```

Comment:
This looks a bit like a switch in disguise. :)

Reply:
I wish TS had "switch expressions" :( https://github.com/concrete-utopia/utopia/pull/5987/commits/2f41b156c6dcdf2e949a4b5cafe105ca917fd162